### DISCUSSION TITLE
Estimated time of arrival for Kotlin 2.0.0 support.

### DISCUSSION THREAD

**Discussion by KarlHakanNordgren** [detekt/detekt#7363] [Q&A] [answered]
I understand it is hard to say exactly, but could you give me some indication of the estimated time of arrival for Kotlin 2.0.0 support.

**Comment by BraisGabin** [comment, marked as answer]
Sure! Detekt already supports kotlin 2.0.0.

**Comment by KarlHakanNordgren** [comment]
@BraisGabin: I'm sorry, I don't understand. The latest version of `detekt` is 1.23.6 which according to [this](https://github.com/detekt/detekt/releases) supports Kotlin 1.9.23. What have I misunderstood?

**Reply by BraisGabin** [reply]
The wording is a bit off. That version of detekt is bundled with kotlin `1.9.23`. But detekt supports any version of kotlin. You can find information about this and common missconfigurations at #6198. The most common one is when using with Spring Boot.

**Reply by KarlHakanNordgren** [reply]
Just to confirm, `detekt` can assess code quality for Kotlin 2.0.0?

**Reply by BraisGabin** [reply]
Yes, it can. There are always the option to have bugs, and if you find something please open an issue because we want to know about those. But the current version of detekt should work perfectly fine with kotlin2.

(Kotlin 2.1 is probably a different story but we are not there yet)

**Reply by KarlHakanNordgren** [reply]
Thanks!
